.github/workflows/2_auto_publish_release.yml

```
@@ -0,0 +1,68 @@
+name: Release stage 2 - auto publish
+
+on:
+  pull_request:  # types AND paths
+    types: [closed]
+    paths: ['cylc/flow/__init__.py']
+    # NOTE: While this is too generic, we use the `if` condition of the job to narrow it down
+    # NOTE: Don't use `branches` as we might create release on any branch
+
+env:
+  # Best not to include the GH token here, only do it for the steps that need it
+  MERGE_SHA: ${{ github.event.pull_request.merge_commit_sha }}
+  CHANGELOG_FILE: CHANGES.md
+
+jobs:
+  publish:
+    if: >-  # NOTE: Can't use top-level env here unfortunately
+      github.event.pull_request.merged == true &&
+      contains(github.event.pull_request.labels.*.name, 'release')
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+
+    - name: Get the version number
+      uses: cylc/release-actions/stage-2/get-version-from-pr@v1
+
+    - name: Checkout repo
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ env.MERGE_SHA }}
+
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Build
+      uses: cylc/release-actions/build-python-package@v1
+
+    - name: Publish distribution to PyPI
+      uses: pypa/gh-action-pypi-publish@v1.4.1
+      with:
+        user: __token__ # uses the API token feature of PyPI - least permissions possible
+        password: ${{ secrets.PYPI_TOKEN }}
```

Can use this to test without risking the cylc-flow package.

```suggestion
        password: ${{ secrets.PYPI_TOKEN }}
        repository_url: https://test.pypi.org/legacy/
```

---

Maybe document this in the file @MetRonnie ?